azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,17 +88,70 @@ protected void handleGet(
+      if (IMAGE_TYPE_NAME_OR_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+        getImageTypeDTOByIdOrImageTypeName(resp, session, templateVariableToValue);
       } else if (GET_IMAGE_TYPE_URI.equals(req.getRequestURI())) {
-        // TODO: Implementation will be provided in the future PR
+        ImageTypeDTO imageTypeDTO;
+        if (req.getQueryString().contains(IMAGE_TYPE)) {
+          final String imageTypeName = HttpRequestUtils
+              .getParam(req, ImageMgmtConstants.IMAGE_TYPE);
+          if (!hasImageManagementPermission(imageTypeName, session.getUser(), Type.GET)) {
+            log.info(FORBIDDEN_USER_ERR_MSG);
+            throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN,
+                FORBIDDEN_USER_ERR_MSG);
+          }
+          imageTypeDTO = this.imageTypeService.findImageTypeWithOwnershipsByName(imageTypeName);
+          sendResponse(resp, HttpServletResponse.SC_OK, imageTypeDTO);
+        }
+        getAllImageTypeDTOs(resp, session);
+      } else {
+        log.info(PATH_NOT_SUPPORTED);
+        throw new ImageMgmtInvalidInputException(ErrorCode.NOT_FOUND, PATH_NOT_SUPPORTED);
+      }
+    } catch (final ImageMgmtInvalidPermissionException e) {
+      log.error("The user provided does not have permissions to access the resource");
+      resp.setStatus(HttpStatus.SC_FORBIDDEN);
+    } catch (final ImageMgmtException e) {
+      if (e.getErrorCode() != null) {
+        log.error("An error has occurred");
+        resp.setStatus(e.getErrorCode().getCode());
+      } else {
+        resp.setStatus(HttpStatus.SC_BAD_REQUEST);
       }
     } catch (final Exception e) {
       log.error("Content is likely not present " + e);
       resp.setStatus(HttpStatus.SC_NOT_FOUND);
     }
   }
 
+  private void getImageTypeDTOByIdOrImageTypeName(
+      final HttpServletResponse resp, final Session session,
+      final Map<String, String> templateVariableToValue) throws IOException {
+    ImageTypeDTO imageTypeDTO;
+    String imageTypeIdOrName = templateVariableToValue.get(IMAGE_TYPE_NAME_OR_ID);
+    if (!StringUtils.isNumeric(imageTypeIdOrName)) {
+
+      imageTypeDTO = this.imageTypeService.findImageTypeWithOwnershipsByName(imageTypeIdOrName);
+    } else {
+      imageTypeDTO = this.imageTypeService.findImageTypeWithOwnershipsById(imageTypeIdOrName);
+    }
+    if (!hasImageManagementPermission(imageTypeDTO.getName(), session.getUser(), Type.GET)) {
+      log.info(FORBIDDEN_USER_ERR_MSG);
+      throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN, FORBIDDEN_USER_ERR_MSG);
+    }
+    sendResponse(resp, HttpServletResponse.SC_OK, imageTypeDTO);
+  }
+
+  private void getAllImageTypeDTOs(final HttpServletResponse resp, final Session session)
+      throws IOException {
+    if (!isAzkabanAdmin(session.getUser())) {
+      throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN, FORBIDDEN_USER_ERR_MSG);
+    }
```

Comment:
Why only Azkaban ADMINs can use this endpoint if non admins users can still make calls to `/imageTypes?&imageType=?`. I would expect that calls to `/imageTypes` by non admin users would return a list of imageTypes  that particular user has access to.

Reply:
I think this is a question for @jakhani. As written it makes sense to me, but I can see it both ways.